src/v/s3/client.h

```diff
@@ -309,6 +311,8 @@ class client_pool : public ss::weakly_referencable<client_pool> {
     configuration _config;
     client_pool_overdraft_policy _policy;
     std::vector<http_client_ptr> _pool;
+    // List of all connections currently used by clients
+    std::list<http_client_ptr> _leased;
```

Comment:
i think the comment here should have a note that says this code depends on the iterator invalidation rules of std::list explicitly. at least, it seems that way, right? above there is an iterator saved in a deferred deleter that can't tolerate invalidation w.r.t. asynchronous inserts?

Reply:
did that